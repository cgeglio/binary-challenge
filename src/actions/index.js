export const login = user => ({
  type: 'LOGIN',
  user
})

export const logout = user => ({
  type: 'LOGOUT',
  user
})

export const addCards = cards => ({
  type: 'ADD_CARDS',
  cards
})

export const removeCards = cards => ({
  type: 'REMOVE_CARDS',
  cards
})

export const addQuestion = question => ({
  type: 'ADD_QUESTION',
  question
})

export const removeQuestion = question => ({
  type: 'REMOVE_QUESTION',
  question
})

export const addFortune = fortune => ({
  type: 'ADD_FORTUNE',
  fortune
})

export const removeFortune = fortune => ({
  type: 'REMOVE_FORTUNE',
  fortune
})

export const addReading = currentReading => ({
  type: 'ADD_READING',
  currentReading
})

export const removeReading = currentReading => ({
  type: 'REMOVE_READING',
  currentReading
})

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
})

export const removeFavorite = favorite => ({
  type: 'REMOVE_FAVORITE',
  favorite
})

export const resetFavorites = favorites => ({
  type: 'RESET_FAVORITES',
  favorites
})
