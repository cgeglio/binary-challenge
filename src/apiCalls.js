export const getCards = () => {
  let proxyUrl = 'https://cors-anywhere.herokuapp.com/',
      targetUrl = 'https://rws-cards-api.herokuapp.com/api/v1/cards/random?n=4'
  return fetch(proxyUrl + targetUrl)
    .then(response => {
      if (!response.ok) {
        throw Error('Error fetching cards')
      }
      return response.json()
    })
}

export const getFortune = () => {
  return fetch('http://fortunecookieapi.herokuapp.com/v1/fortunes/')
    .then(response => {
      if (!response.ok) {
        throw Error('Error fetching fortune')
      }
      return response.json()
    })
}
