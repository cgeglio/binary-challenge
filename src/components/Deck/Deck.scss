@import "../../mixins.scss";

.deck {
  align-items: center;
  background-color: black;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: auto;
  justify-content: space-around;
  padding: 0 5%;
}

.deck-title {
  font-family: 'Croissant One';
  font-size: 1.5em;
  text-align: center;
  width: 100%;
}

.btn-container {
  display: flex;
  height: 30px;
  justify-content: center;
  padding: 20px;
  width: 100%;
}

.btn-container button {
  margin-top: 0;
  position: relative;
}
