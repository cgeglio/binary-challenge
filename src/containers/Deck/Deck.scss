@import "../../mixins.scss";
@import "../../variables.scss";

.deck {
  align-items: center;
  background-color: black;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: auto;
  justify-content: space-around;
  padding: 0 5%;
}

.deck-title {
  font-family: $primary-font;
  font-size: 1.5em;
  text-align: center;
  width: 100%;
}

.btn-container {
  display: flex;
  height: 30px;
  justify-content: center;
  padding: 20px;
  width: 100%;
}

.btn-container button {
  margin-top: 0;
  position: relative;
}
