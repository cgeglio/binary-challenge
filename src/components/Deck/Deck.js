import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Loader from '../../components/Loader/Loader';
import './Deck.scss';
import { getDeck } from '../../apiCalls';
import Card from '../Card/Card'

class Deck extends Component {
  constructor() {
    super();
    this.state={deck: []}
  }

  componentDidMount() {
    getDeck()
      .then(deck => this.setState({deck: deck.cards}))
      .catch(error => console.log(error))
  }

  render() {
    return (
      !this.state.deck.length ? <Loader /> :
      <section className='deck'>
        <h2 className='deck-title'>The Tarot Deck</h2>
        <div className='btn-container'><Link to='/home'><button className='back-btn'>Home</button></Link></div>
        {this.state.deck.map(card => {
          return <Card key={card.name_short} card={card} />
        })}
      </section>
    )
  }
}

export default Deck;
