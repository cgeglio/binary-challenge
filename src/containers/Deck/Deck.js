import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Loader from '../../components/Loader/Loader';
import './Deck.scss';
import { getDeck } from '../../apiCalls';
import Card from '../../components/Card/Card';
import { removeQuestion } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


export class Deck extends Component {
  constructor() {
    super();
    this.state={deck: []}
  }

  componentDidMount() {
    getDeck()
      .then(deck => this.setState({deck: deck.cards}))
      .catch(error => console.log(error))
  }

  render() {
    return (
      !this.state.deck.length ? <Loader /> :
      <section className='deck'>
        <h2 className='deck-title'>The Tarot Deck</h2>
        <div className='btn-container'><Link to='/home'><button onClick={() => this.props.resetQuestion('question')} className='back-btn'>Home</button></Link></div>
        {this.state.deck.map(card => {
          let number = Math.floor(1 + Math.random() * 4);
          return <Card key={card.name_short+card.name_short} card={card} number={number}/>
        })}
      </section>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  resetQuestion: question => (dispatch(removeQuestion(question)))
})

export default connect(null, mapDispatchToProps)(Deck);

Deck.propTypes = {
  resetQuestion: PropTypes.func
}
