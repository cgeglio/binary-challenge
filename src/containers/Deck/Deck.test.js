import React from 'react';
import { Deck, mapDispatchToProps } from './Deck';
import { removeQuestion } from '../../actions';
import { shallow } from 'enzyme';
import { getDeck } from '../../apiCalls';
import { mockRandom } from 'jest-mock-random';

jest.mock('../../apiCalls');

describe('Deck', () => {

  let mockResetQuestion;
  let wrapper;

  beforeEach(() => {
    mockResetQuestion = jest.fn().mockImplementation();
    mockRandom([.01]);

    getDeck.mockImplementation(() => {
      return Promise.resolve({cards: [
        {name: 'The Sun', value: 8},
        {name: 'The Hermit', value: 12}
      ]});
    });

    wrapper = shallow(<Deck resetQuestion={mockResetQuestion} />);
  });

  describe('Deck container/component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call resetQuestion when the home button is clicked', () => {
      wrapper.find('.back-btn').simulate('click');
      expect(mockResetQuestion).toHaveBeenCalledWith('question');
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the removeQuestion action when resetQuestion is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeQuestion('question');
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.resetQuestion('question');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
