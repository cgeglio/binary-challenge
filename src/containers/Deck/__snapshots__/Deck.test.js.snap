// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck Deck container/component should match the snapshot 1`] = `
<section
  className="deck"
>
  <h2
    className="deck-title"
  >
    The Tarot Deck
  </h2>
  <div
    className="btn-container"
  >
    <Link
      to="/home"
    >
      <button
        className="back-btn"
        onClick={[Function]}
      >
        Home
      </button>
    </Link>
  </div>
  <Card
    card={
      Object {
        "name": "The Sun",
        "value": 8,
      }
    }
    id="on-flip"
    key="NaN"
  />
  <Card
    card={
      Object {
        "name": "The Hermit",
        "value": 12,
      }
    }
    id="on-flip"
    key="NaN"
  />
</section>
`;
